crates/BUILD

```diff
@@ -0,0 +1,76 @@
+# Licensed to the Apache Software Foundation (ASF) under one
+# or more contributor license agreements.  See the NOTICE file
+# distributed with this work for additional information
+# regarding copyright ownership.  The ASF licenses this file
+# to you under the Apache License, Version 2.0 (the
+# "License"); you may not use this file except in compliance
+# with the License.  You may obtain a copy of the License at
+#
+#   http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing,
+# software distributed under the License is distributed on an
+# "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+# KIND, either express or implied.  See the License for the
+# specific language governing permissions and limitations
+# under the License.
+#
+
+exports_files(["rules.bzl"])
+
+load("@bazel_skylib//:bzl_library.bzl", "bzl_library")
+load("@io_bazel_rules_kotlin//kotlin:kotlin.bzl", "kt_jvm_library")
+
+bzl_library(
+    name = "lib",
+    srcs = [
+        "rules.bzl",
+    ],
+    visibility = ["//visibility:public"]
+)
+
+
+kt_jvm_library(
+    name = "crate-assembler-lib",
+    deps = [
+      "@maven//:com_eclipsesource_minimal_json_minimal_json",
+      "@maven//:info_picocli_picocli",
+      "@maven//:org_apache_commons_commons_compress",
+      "@maven//:com_fasterxml_jackson_core_jackson_databind",
+      "@maven//:com_fasterxml_jackson_dataformat_jackson_dataformat_toml",
+    ],
+    srcs = [
+        "CrateAssembler.kt"
+    ]
+)
+
+java_binary(
+    name = "crate-assembler",
+    runtime_deps = [
+        ":crate-assembler-lib"
+    ],
+    main_class = "CrateAssemblerKt",
+    visibility = ["//visibility:public"]
+)
+
+
+kt_jvm_library(
+    name = "crate-deployer-lib",
+    deps = [
+      "@maven//:com_eclipsesource_minimal_json_minimal_json",
+      "@maven//:info_picocli_picocli",
+      "@maven//:com_google_http_client_google_http_client",
+    ],
+    srcs = [
+        "CrateDeployer.kt"
+    ]
+)
+
+java_binary(
+    name = "crate-deployer",
```

Comment:
This target is needed for the `deploy_crate` rule to be able to use a fat JAR with all dependencies included: `@vaticle_bazel_distribution//crates:crate-deployer_deploy.jar`